homework/vector-of-shared-ptrs/vectorFunctions.cpp

```diff
@@ -0,0 +1,39 @@
+#include <iostream>
+#include <vector>
+#include <memory>
+
+std::vector<std::shared_ptr<int>> generate(int count) {
+    std::vector<std::shared_ptr<int>>vec;
+    vec.reserve(count);
+    for (int i = 0; i < count; i++){
+        vec.push_back(std::make_shared<int>(i));
+    }
+    return vec;
+}
+
+void print(std::vector<std::shared_ptr<int>> vec) {
+    for (auto i : vec){
+        std::cout << i << "\n";
+    }
+}
+
+void add10(std::vector<std::shared_ptr<int>> vec) {
+    for (auto i : vec){
+        if (i != nullptr){   
+            *i += 10;
```

Comment:
if (value) is enough. This will call implicit `bool` operator which compare this ptr with `nullptr`